starter/README.md

```diff
@@ -1,20 +1,18 @@
 # Base Starter for Vaadin Flow and OSGi in NPM mode
 
-Servlet initial parameters may be passed via `@Component` annotation parameter `property`, e.g.
+Vaadin Servlet initial parameters may be passed via `@Component` annotation parameter `property`, e.g.
 
 ```java
 @Component(immediate = false, service = Servlet.class, property = {
   HttpWhiteboardConstants.HTTP_WHITEBOARD_SERVLET_INIT_PARAM_PREFIX +
-  InitParameters.SERVLET_PARAMETER_COMPATIBILITY_MODE + "=false"}) 
+  InitParameters.SERVLET_PARAMETER_PRODUCTION_MODE + "=false"})
 ```
 or via some annotation like `VaadinMode` which is available in the example.
 
-Same way can be used to set production mode via the initial parameters.
-Another way to set these values is explicitly set them via the vaadin maven plugin configuration:
+Another way to set the property values is explicitly set them via the vaadin maven plugin configuration:
```

Comment:
This is inaccurate.
There was a reason why almost everything has been said exactly as it has been said.
Not _ANY_ property value can be set via token file.
There is a limited number of _MAVEN_ configuration parameters which are reflected to the _SAME_ servlet init property name.
`productionMode`  is mapped exactly as is : the same name is init parameter and the same maven config parameter can be used  in the pom.
There are config maven parameters which has different name comparing to init parameters.
And there are a lot servlet init parameters and only few configuration parameters.

All those things are written in the generic Vaadin docs.
And there is no reason to repeat it here. Since it not Vaadin tutorial.
This is just a short reminder how you may use most important parameters.

So the sentence should not say `property values`  it should say exactly about `prodctionMode` and nothing else.
compatibility mode was also the parameter name whose name has 1:1 mapping.
May be it's worth to mention that I said above to avoid confusion. But this is not a tutorial, this is just a short readme. So I'm not sure.

Reply:
Thanks! I've mentioned only production mode here.

I couldn't find above explanations in Vaadin generic docs (vaadin.com/docs and on a new beta version).
So it's worth to mention it here.
With these explanations it's much easier to understand how the parameters are handled.

I've spent some time exploring how the properties from vaadin maven plugin configuration go to `flow-build-info.json` file and them how they are merged with other init parameters.

I'm not sure this part is described well in the docs, at least I couldn't find it.
This is I found https://vaadin.com/docs/v18/flow/advanced/tutorial-flow-runtime-configuration.html#tutorial-all-vaadin-properties
+ all Vaadin properties chapter https://vaadin.com/docs/v18/flow/advanced/tutorial-flow-runtime-configuration.html#tutorial-all-vaadin-properties.

Reply:
There is not exact explanation of matching between init parameters and maven config parameters.

There is explanation for init parameters : what each parameter means.
There is explanation for maven config parameter: what each parameter means.

From these two lists you may find how to match init parameter to maven config parameter.

May be we should have a section which does this matching out of the box .
But then there will be several questions:
- where to put this section ? (servlet/maven ? or totally different)
- how to find this section ? (related above)
- every time it needs to be updated.

Anyway: this is out of the scope of this PR. Feel free to create a ticket about this in docs.

I would not give so detailed explanation about parameters: this is out of scope of this repo.
it should gives just short explanation how to do things. And warning about matching properties should be as short as possible: it's not Vaadin tutorial . It describes how to configure things for OSGi.
And maven config parameter is mentioned here just with one purpose : it allowed to specify compat mode (and production mode) without having any code in Java.
compat mode is now out of the scope.
production mode is quite important parameter.
Everything else should be done according to the generic docs.

Reply:
> Anyway: this is out of the scope of this PR. Feel free to create a ticket about this in docs.

Sure, good to have this section some day. I'll make a ticket for that.

> I would not give so detailed explanation about parameters: this is out of scope of this repo.
it should gives just short explanation how to do things. And warning about matching properties should be as short as possible: it's not Vaadin tutorial . It describes how to configure things for OSGi.

True. But I'm not certain how to cut it to be more shorter and to not lose the idea. Anyway, I propose to keep it as is, because it always useful to have an instant explanations right in the starter especially when someone use it for the first time, even though these explanations repeats the generic docs.